crates/sui-core/src/authority_server.rs

```diff
@@ -327,6 +327,27 @@ impl ValidatorService {
                 "Cannot execute system certificate via RPC interface! {certificate:?}"
             )));
         }
+        for (object_id, queue_len) in state.transaction_manager().objects_queue_len(
+            certificate
+                .data()
+                .intent_message
+                .value
+                .kind
+                .input_objects()?
+                .into_iter()
+                .map(|r| r.object_id())
+                .collect(),
+        ) {
+            // When this occurs, most likely transactions piled up on a shared object.
+            if queue_len >= MAX_PER_OBJECT_EXECUTION_QUEUE_LENGTH {
+                return Err(SuiError::TooManyTransactionsPendingOnObject {
+                    object_id,
+                    queue_len,
+                    threshold: MAX_PER_OBJECT_EXECUTION_QUEUE_LENGTH,
+                }
+                .into());
+            }
+        }
```

Comment:
what's the rationale for blocking certificate execution? if one validator accepts the tx, it will arrive via consensus anyway.

Reply:
This blocks submission via RPC. The rationale is that when all validators are overloaded, this would slow down submitting the certificate to Narwhal as well. It is true that a faster or byzantine validator can bypass this, which is why I mentioned the scheme is not bulletproof. But I think this is still marginally useful. LMK if you feel strongly against it.

Reply:
One case is that a client first signs 100k transactions, then try to submit them all at once. A robust scheme to protect against this, considering byzantine validators, would be a bit involved. The simplest way seems to be slowing down submissions for now.